### PR TITLE
ENH: add convenience blt_list_append macro

### PR DESCRIPTION
**Pull request by gzagaris**
This commit adds a blt_list_append() macro, which is
essentially a wrapper around CMake's list(APPPEND...)
that allows to inline a conditional check for convenience
and clarity.

As an example, code that looks like this:
```
set(mylist)

if ( ${USE_FOO} )
  list(APPEND mylist foo)
endif()

if ( ${USE_BAR} )
  list(APPEND mylist bar)
endif()
```

Can be simplified using the blt_list_append() macro like this:
```
set(mylist)
blt_list_append( TO mylist ELEMENTS foo IF ${USE_FOO} )
blt_list_append( TO mylist ELEMENTS bar IF ${USE_BAR} )
```

### CODE REVIEW

File: cmake/BLTMacros.cmake [hunk 1]
```diff
@@ -43,6 +43,52 @@
...
+##
+## Appends elements to a list if the specified bool evaluates to true.
+##
+## This macro is essentially a wrapper around CMake's `list(APPEND ...)`
+## command which allows inlining a conditional check within the same call
+## for clarity and convenience.
+##
+## This macro requires specifying:
+##   (1) The target list to append to by passing TO <list>
+##   (2) A condition to check by passing IF <bool>
```
Comment (white238) [inline cmake/BLTMacros.cmake]: Please make a note that the IF argument cannot handle anything other than a single variable.  For example, the following won't work:

blt_list_append( TO mylist ELEMENTS C
                            IF NOT ${ENABLE_C} )

Comment (gzagaris) [reply cmake/BLTMacros.cmake]: Yes, good point. I'll add a comment. 

As a side note, we might be able to make use of the boolean generator expression $<BOOL: NOT ${ENABLE_C}> to inline evaluate a boolean expression and pass the result to the macro. I'll give this a shot and add comments accordingly.

Comment (kennyweiss) [reply cmake/BLTMacros.cmake]: I like that idea

Comment (white238) [reply cmake/BLTMacros.cmake]: Another option is to add an IF_NOT. Though this is a pretty slippery slope. IF_AND, IF_OR.. where does it stop.

Comment (gzagaris) [reply cmake/BLTMacros.cmake]: Yeah, it's a rabbit hole. Unfortunately, the `$<BOOL: ...>` doesn't work in this context either. I added a clarification to the documentation for the macro.

**Pull request merged**